File: tools/misc/launchDev/writeVersion.ts
import { v7 as uuidv7 } from "uuid";
import { writeVersion } from "../update/version.ts";
import { writeBuildid2 } from "../update/buildid2.ts";
import { resolve } from "pathe";
import { ensureDir } from "../../utils/utils.ts";

// Generate version and build ID
export async function genVersion() {
  const projectRoot = resolve(import.meta.dirname!, "../../../");
  await writeVersion(resolve(projectRoot, "static/gecko"));
  try {
    await Deno.stat("_dist");
  } catch {
    await ensureDir("_dist");
  }
  await writeBuildid2(resolve(projectRoot, "_dist"), uuidv7());
}

File: tools/misc/launchDev/writeVersion.rb
require 'fileutils'
require 'securerandom'

def write_version(dir)
  File.write(File.join(dir, 'version.txt'), "version: 1.0.0\n")
end

def write_buildid2(dir, buildid)
  File.write(File.join(dir, 'buildid2.txt'), "buildid2: #{buildid}\n")
end

def gen_version
  root = File.expand_path('../../../', __dir__)
  gecko = File.join(root, 'static/gecko')
  dist = File.join(root, '_dist')
  FileUtils.mkdir_p(dist)
  write_version(gecko)
  write_buildid2(dist, SecureRandom.uuid)
end

gen_version if $PROGRAM_NAME == __FILE__